src/metaprob/state.clj

```diff
@@ -132,23 +134,32 @@
        (= (count tr) 1)
        (contains? tr :value)))
 
-;; Convert hash-map to heterogeneous canonical clojure form
+;; Convert hash-map to heterogeneous canonical clojure form.
+
+;; I'm sorry I failed to record the reason that the 'don't be lazy'
+;; command is there; there must have been a failure at some point that 
+;; I attributed to laziness in these maps.
 
 (defn map-to-state [m]
+  (doseq [entry m] true)    ;Don't be lazy!
```

Comment:
I believe the only place laziness shows up in Clojure is in [lazy sequences](http://clojure-doc.org/articles/language/laziness.html).

Reply:
I agree - I think this code is vestigial, and precedes some reorganization of the function that makes it moot.  So I should probably delete it.  Thanks for noticing